src/shared/GitLab.UI/ViewModels/CredentialsViewModel.cs

```diff
@@ -0,0 +1,181 @@
+using System.ComponentModel;
+using System.Windows.Input;
+using GitCredentialManager;
+using GitCredentialManager.UI;
+using GitCredentialManager.UI.ViewModels;
+
+namespace GitLab.UI.ViewModels
+{
+    public class CredentialsViewModel : WindowViewModel
+    {
+        private readonly IEnvironment _environment;
+
+        private string _url;
+        private string _token;
+        private string _tokenUserName;
+        private string _userName;
+        private string _password;
+        private bool _showBrowserLogin;
+        private bool _showTokenLogin;
+        private bool _showBasicLogin;
+        private ICommand _signUpCommand;
+        private ICommand _signInBrowserCommand;
+        private RelayCommand _signInBasicCommand;
+        private RelayCommand _signInTokenCommand;
+
+        public CredentialsViewModel()
+        {
+            // Constructor the XAML designer
+        }
+
+        public CredentialsViewModel(IEnvironment environment)
+        {
+            EnsureArgument.NotNull(environment, nameof(environment));
+
+            _environment = environment;
+
+            Title = "Connect to GitLab";
+            SignUpCommand = new RelayCommand(SignUp);
+            SignInBrowserCommand = new RelayCommand(SignInBrowser);
+            SignInTokenCommand = new RelayCommand(SignInToken, CanSignInToken);
+            SignInBasicCommand = new RelayCommand(SignInBasic, CanSignInBasic);
+
+            PropertyChanged += OnPropertyChanged;
+        }
+
+        private void OnPropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            switch (e.PropertyName)
+            {
+                case nameof(UserName):
+                case nameof(Password):
+                    SignInBasicCommand.RaiseCanExecuteChanged();
+                    break;
+
+                case nameof(Token):
+                    SignInTokenCommand.RaiseCanExecuteChanged();
+                    break;
+            }
+        }
+
+        private void SignUp()
+        {
+            BrowserUtils.OpenDefaultBrowser(_environment, "https://about.gitlab.com/");
```

Comment:
URL should depend on host, eg. https://gitlab.example.com/users/sign_up or simply https://gitlab.example.com/